### DISCUSSION TITLE
Profile page

### DISCUSSION THREAD

**Discussion by sunehs** [elmadev/elmaonline-site#420] [Feature discussion]
For general discussion of the profile page: https://elma.online/kuskis/Kopaka

This post will be updated to contain planned features and changes as well as ideas that needs to be fledged out. Discuss them below. If you want to work on one of these, create an issue if it doesn't exist and assign yourself.

**Planned**
- [ ] Improve TT load time: Possibly using cached data from levelpack page

**Ideas**
- Customizable: For own page pick which pack to show tt from in header for example
- Show ranking increase in latest battles
- Show records in select levelpacks
- Filter battles by battles won and maybe other things
- Historic ranking, graph etc.

**Comment by j-masci** [comment]
Ideas
- Show all world records held by player. Would require some form of de-normalized data. if [levelStats](https://github.com/elmadev/elmaonline-site/issues/410) implemented, then becomes easy.
- Show battles won (not just battles played).
- Player ranking over time graph or something? (low priority) I think the ranking history data is there.

**Comment by j-masci** [comment]
Using new LevelStats data:

- Show the number of world records (and/or 1st, 2nd, 3rd place finishes) somewhere on profile page (maybe near top).
- List all world records by date driven. Can also show all 2nd, 3rd, and up to 10th place finishes if there's a good way to do it (choose a number between 1 and 10 and data gets updated?).
